radiation/python/radsw/stencils_sw_gt4py.py

```diff
@@ -0,0 +1,4107 @@
+                                    + fint
+                                    * (
+                                        ssaice3[0, 0, 0][index + 1, ib8]
+                                        - ssaice3[0, 0, 0][index, ib8]
+                                    ),
+                                ),
+                            )
+                            asycoice = max(
+                                0.0,
+                                min(
+                                    1.0,
+                                    asyice3[0, 0, 0][index, ib8]
+                                    + fint
+                                    * (
+                                        asyice3[0, 0, 0][index + 1, ib8]
+                                        - asyice3[0, 0, 0][index, ib8]
+                                    ),
+                                ),
+                            )
+
+                            tauice[0, 0, 0][ib8] = cldice * extcoice
+                            ssaice[0, 0, 0][ib8] = tauice[0, 0, 0][ib8] * ssacoice
+                            asyice[0, 0, 0][ib8] = ssaice[0, 0, 0][ib8] * asycoice
+
+                for ib9 in range(nbdsw):
+                    jb = nblow + ib9 - 16
+                    taucw[0, 0, 0][ib9] = (
+                        tauliq[0, 0, 0][jb] + tauice[0, 0, 0][jb] + tauran + tausnw
+                    )
+                    ssacw[0, 0, 0][ib9] = (
+                        ssaliq[0, 0, 0][jb]
+                        + ssaice[0, 0, 0][jb]
+                        + ssaran[0, 0, 0][jb]
+                        + ssasnw[0, 0, 0][jb]
+                    )
+                    asycw[0, 0, 0][ib9] = (
+                        asyliq[0, 0, 0][jb]
+                        + asyice[0, 0, 0][jb]
+                        + asyran[0, 0, 0][jb]
+                        + asysnw[0, 0, 0][jb]
+                    )
+        else:
+            if cfrac > ftiny:
+                for ib10 in range(nbdsw):
+                    taucw[0, 0, 0][ib10] = cdat1
+                    ssacw[0, 0, 0][ib10] = cdat1 * cdat2
+                    asycw[0, 0, 0][ib10] = ssacw[0, 0, 0][ib10] * cdat3
+
+        # if physparam::isubcsw > 0, call mcica_subcol() to distribute
+        # cloud properties to each g-point.
+
+        if isubcsw > 0:
+            cldf = 0.0 if cfrac < ftiny else cfrac
+
+    # This section builds mcica_subcol from the fortran into cldprop.
+    # Here I've read in the generated random numbers until we figure out
+    # what to do with them. This will definitely need to change in future.
+    # Only the iovrlw = 1 option is ported from Fortran
+    with computation(PARALLEL), interval(2, None):
+        tem1 = 1.0 - cldf[0, 0, -1]
+
+        for n in range(ngptsw):
+            if cdfunc[0, 0, -1][n] > tem1:
+                cdfunc[0, 0, 0][n] = cdfunc[0, 0, -1][n]
+            else:
+                cdfunc[0, 0, 0][n] = cdfunc[0, 0, 0][n] * tem1
+
```

Comment:
I think this probably shouldn't be a PARALLEL computation, because it is both reading cdfunc[0,0,-1] and writing to cdfunc[0,0,0]. The validation may have gotten lucky, but with the vertical axis run in parallel you aren't guarenteel the layer at [0,0,-1] hasn't been changed before [0,0,0] is set.

Reply:
Yes, I've run into this elsewhere. I will change it